src/charm.py

```diff
@@ -112,6 +114,7 @@ def _check_tokens_integration(self, event) -> bool:
             status.add(WaitingStatus(evaluation))
         return False
 
+    @status.on_error(ops.WaitingStatus("Waiting on CNI"))
```

Comment:
Are any of these methods raising exceptions? I tried to look up each one, but I think all of them are quite safe, unless you don't have the proper CNI related, right? Is that the case we are trying to catch here?

Reply:
if the CNI is related, but no data is yet available -- they could fail with a KeyError

that's what was going on in the unit tests